### DISCUSSION TITLE
Download Data Failed

### DISCUSSION THREAD

**Discussion by KeithAmbrose** [NightscoutFoundation/xDrip#3317] [Q&A]
Hi there, I haven't been able to get xDrip to work on my new phone. I recently upgraded to the pixel 8 pro and I have been using extra for some time on my pixel 6 with a lot of success. I cannot get past the download data failed step and was hoping that the community might be able to help me out. I have a feeling this is something simple, a box that I've not checked or something. 

I have the most recent version of xdrip and the most recent version of CareLink. Running Android 14. 

![Screenshot_20240126-143928](https://github.com/NightscoutFoundation/xDrip/assets/137965905/238b8c30-f69c-4c12-a686-d46c54fd6be9)

**Comment by PhilTU** [comment]
what is the hardware data source ?
